### PR TITLE
Bugfix/box shadow

### LINKED ISSUE #59: boxShadow transform does not support rbg, rgba, hsl or hsla colors
Currently `box-shadow` transform does not support `rgb`, `rgba`, `hsl` or `hsla` colors. 

Here are a few test cases that are failing:

```js
it('transforms box-shadow with rgb color', () => runTest([
  ['box-shadow', '10px 20px rgb(100, 100, 100)'],
], {
  shadowOffset: { width: 10, height: 20 },
  shadowRadius: 0,
  shadowColor: 'rgb(100, 100, 100)',
}));

it('transforms box-shadow with rgba color', () => runTest([
  ['box-shadow', '10px 20px rgba(100, 100, 100, 0.5)'],
], {
  shadowOffset: { width: 10, height: 20 },
  shadowRadius: 0,
  shadowColor: 'rgba(100, 100, 100, 0.5)',
}));

it('transforms box-shadow with hsl color', () => runTest([
  ['box-shadow', '10px 20px hsl(120, 100%, 50%)'],
], {
  shadowOffset: { width: 10, height: 20 },
  shadowRadius: 0,
  shadowColor: 'hsl(120, 100%, 50%)',
}));

it('transforms box-shadow with hsla color', () => runTest([
  ['box-shadow', '10px 20px hsla(120, 100%, 50%, 0.7)'],
], {
  shadowOffset: { width: 10, height: 20 },
  shadowRadius: 0,
  shadowColor: 'hsla(120, 100%, 50%, 0.7)',
}));
```

In addition to that, `shadowOpacity` property set to value `1` should probably be added by default to make sure that the shadow is always visible. I tested it, and I can confirm that the opacity for the shadow can be controlled by using `rgba` or `hsla` color's alpha value, so there is no need for `shadowOpacity` to be something else than `1`.

@jacobp100 @mxstbr

### PR DESCRIPTION
**Pull request by kristerkari**
fixes #59

### CODE REVIEW

File: src/transforms/boxShadow.js [hunk 1]
```diff
@@ -28,10 +28,8 @@ module.exports = (tokenStream) => {
         tokenStream.expect(SPACE);
         blurRadius = tokenStream.expect(LENGTH);
       }
-    } else if (color === undefined && (
-      tokenStream.matchesFunction() || tokenStream.matches(WORD)
-    )) {
-      color = String(tokenStream.lastValue);
+    } else if (tokenStream.matches(COLOR)) {
```
Comment (jacobp100) [inline src/transforms/boxShadow.js]: I think we need to check `color === undefined` to stop `box-shadow: 0 0 0 red yellow green blue`

Comment (kristerkari) [reply src/transforms/boxShadow.js]: ok, should I add a test case for that?

Comment (kristerkari) [reply src/transforms/boxShadow.js]: I added a test case and a fix for that. :)

**Comment by mxstbr** [comment]
Thank you so much for helping us improve styled-components! Based on our [Community Guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md) every person that has a PR of any kind merged is offered **an invitation to the styled-components organization**—that is you right now!

Should you accept, you'll get write access to the main repository. (and a fancy `styled-components` logo on your GitHub profile!) You'll be able to label and close issues, merge pull requests, create new branches, etc. We want you to help us out with those things, so don't be scared to do them! Make sure to read our [contribution and community guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md), which explains all of this in more detail. Of course, if you have any questions just let me know!

**Comment by jacobp100** [comment]
Perfect! Thanks a lot! 😀

**Comment by kristerkari** [comment]
Great :) Would it be possible to get a new patch release with these changes?

**Comment by jacobp100** [comment]
I’ll do it tonight. I wanna get some token stream perf improvements in too

**Pull request merged**